pyuvdata/utils.py

```diff
@@ -2071,3 +2072,256 @@ def apply_uvflag(
 
     if not inplace:
         return uvd
+
+
+def parse_ants(uv, ant_str, print_toggle=False, x_orientation=None):
+    """
+    Get antpair and polarization from parsing an aipy-style ant string.
+
+    Used to support the the select function.
+    Generates two lists of antenna pair tuples and polarization indices based
+    on parsing of the string ant_str.  If no valid polarizations (pseudo-Stokes
+    params, or combinations of [lr] or [xy]) or antenna numbers are found in
+    ant_str, ant_pairs_nums and polarizations are returned as None.
+
+    Parameters
+    ----------
+    uv : UVBase Object
+        A UVBased object that supports the following functions and parameters:
+         - get_ants
+         - get_antpairs
+         - get_pols
+         - x_orientation
```

Comment:
I just noticed this one this morning, but not sure the best way to handle it. Currently the function is assuming the object has an attribute `x_orientation` to handle the polarization conventions, but I also added this keyword for x_orientation which is never used. Not sure which is the preferred way to handle x_orientation for this function.

Reply:
`x_orientation` is not a required attribute on the objects, so I don't think you can assume it's set. In the case that it's not set, allowing someone to set it in this function is nice. I can see 2 options:

1.) only use the passed in parameter if the attribute is not set on the object (so the object's attribute overrides the parameter).
2.) Use the parameter regardless of what is set on the object (so that it overrides the object's attribute).

I think either is ok, but you need to be clear about it in the docstring.

Reply:
I think 2 is more inline with how some other function will operate. If it is set on an object, it should be passed as the parameter to the function.

Reply:
I implemented option 2. I believe.